src/Tools/vmOpsTool/VMWareImpl.java

```diff
@@ -608,7 +608,8 @@ private ManagedObjectReference findSnapshotInTree(List<VirtualMachineSnapshotTre
 
             if (vmSnapshot.getName().equalsIgnoreCase(snapshotName)) {
                 System.out.println(String.format("Found snapshot [ %s ] for virtual machine.", snapshotName));
-                return vmSnapshot.getSnapshot();
+                snapshotMor =  vmSnapshot.getSnapshot();
+				break;
             } else {
                 List<VirtualMachineSnapshotTree> childTree = vmSnapshot.getChildSnapshotList();
                 snapshotMor = findSnapshotInTree(childTree, snapshotName);

```

Comment:
We should break loop here instead above return statement. condition would be if(snapshotMor != null)
{
 break;
}